samples/sensor/accel_polling/src/main.c

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2022 TOKITA Hiroshi <tokita.hiroshi@fujitsu.com
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <drivers/sensor.h>
+#include <zephyr.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+static const struct device *sensors[] = {
+#if DT_NODE_EXISTS(DT_ALIAS(accel0))
+	DEVICE_DT_GET(DT_ALIAS(accel0)),
+#endif
+#if DT_NODE_EXISTS(DT_ALIAS(accel1))
+	DEVICE_DT_GET(DT_ALIAS(accel1)),
+#endif
+#if DT_NODE_EXISTS(DT_ALIAS(accel2))
+	DEVICE_DT_GET(DT_ALIAS(accel2)),
+#endif
+#if DT_NODE_EXISTS(DT_ALIAS(accel3))
+	DEVICE_DT_GET(DT_ALIAS(accel3)),
+#endif
+#if DT_NODE_EXISTS(DT_ALIAS(accel4))
+	DEVICE_DT_GET(DT_ALIAS(accel4)),
+#endif
+#if DT_NODE_EXISTS(DT_ALIAS(accel5))
+	DEVICE_DT_GET(DT_ALIAS(accel5)),
+#endif
+#if DT_NODE_EXISTS(DT_ALIAS(accel6))
+	DEVICE_DT_GET(DT_ALIAS(accel6)),
+#endif
+#if DT_NODE_EXISTS(DT_ALIAS(accel7))
+	DEVICE_DT_GET(DT_ALIAS(accel7)),
+#endif
+#if DT_NODE_EXISTS(DT_ALIAS(accel8))
+	DEVICE_DT_GET(DT_ALIAS(accel8)),
+#endif
+#if DT_NODE_EXISTS(DT_ALIAS(accel9))
+	DEVICE_DT_GET(DT_ALIAS(accel9)),
+#endif
+};
+
+static const enum sensor_channel channels[] = {
+	SENSOR_CHAN_ACCEL_X,
+	SENSOR_CHAN_ACCEL_Y,
+	SENSOR_CHAN_ACCEL_Z,
+};
+
+void format_sensor_value(struct sensor_value *value, char *buf, size_t len)
+{
+	if (value->val1 == 0) {
+		snprintf(buf, len, "  %c0.%06u", (value->val2 < 0 ? '-' : ' '), abs(value->val2));
+		return;
+	}
+	snprintf(buf, len, "%4d.%06u", value->val1, abs(value->val2));
+}
+
+static int print_accels(const struct device *dev)
+{
+	int ret;
+	struct sensor_value accel;
+	const size_t buffer_length = 16;
+	char strvalue[ARRAY_SIZE(channels)][buffer_length];
+
+	ret = sensor_sample_fetch(dev);
+	if (ret < 0) {
+		printk("%s: sensor_sample_fetch() failed: %d\n", dev->name, ret);
+		return ret;
+	}
+
+	for (size_t i = 0; i < ARRAY_SIZE(channels); i++) {
+		ret = sensor_channel_get(dev, channels[i], &accel);
+		if (ret < 0) {
+			printk("%s: sensor_channel_get(SENSOR_CHAN_ACCEL_%c) failed: %d\n",
+			       dev->name, 'X' + i, ret);
+			return ret;
+		}
+		format_sensor_value(&accel, strvalue[i], buffer_length);
+	}
+
+	printk("%16s [m/s^2]:    (%s, %s, %s)\n", dev->name, strvalue[0], strvalue[1], strvalue[2]);
+
+	return 0;
+}
+
+void main(void)
+{
+	int ret;
+
+	for (uint32_t i = 0; i < ARRAY_SIZE(sensors); i++) {
+		if (!device_is_ready(sensors[i])) {
+			printk("sensor: device %s not ready.\n", sensors[i]->name);
+			return;
+		}
+	}
+
+	while (1) {
+		for (uint32_t i = 0; i < ARRAY_SIZE(sensors); i++) {
+			ret = print_accels(sensors[i]);
+			if (ret < 0) {
+				return;
+			}
+			k_sleep(K_MSEC(1000/ARRAY_SIZE(sensors)));
```

Comment:
why add sleep between different accelerometer reads and not a `k_sleep(K_MSEC(1000))` after all are processed?

Reply:
No strong reason is there.
Fix to fetch data from all sensors wit 1s interval.